tar_scm.py

```diff
@@ -761,6 +761,52 @@ def detect_changes_commands_git(repodir, subdir, changes):
     return changes
 
 
+def detect_changes_commands_svn(repodir, changes):
+    """Detect changes between GIT revisions."""
+    last_rev = changes['revision']
+    new_lines = []
+
+    if last_rev is None:
+        last_rev = safe_run(['svn', 'log', '-l10', '-q', '--incremental'], cwd=repodir)[1].split('\n')
+        # remove blank entry on end
+        last_rev.pop()
+        # return last entry
+        last_rev = last_rev[-1]
+        # retrieve the revision number
+        last_rev = re.search(r'^r[0-9]*', last_rev, re.M).group()
+        last_rev = last_rev.replace("r", "")
+        logging.debug("First run get log for initial release")
+        lines = safe_run(['svn', 'log', '-r', '%s' % last_rev, '--xml'], repodir)[1]
+        lines = re.findall(r"<msg>.*?</msg>", lines, re.S)
+        for line in lines:
+            line = line.replace("<msg>", "").replace("</msg>", "")
+            logging.info("Line: %s" % line)
+            new_lines = new_lines + line.split("\n")
+
+    current_rev = safe_run(['svn', 'log', '-l1', '-q', '--incremental'], cwd=repodir)[1].split('\n')
```

Comment:
I think it would be worth extracting calculation of `current_rev` into a separate function, since this one is currently too long.